lisa/operating_system.py

```diff
@@ -150,15 +214,65 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _package_exists(self, package: str, signed: bool = True) -> bool:
         raise NotImplementedError()
 
     def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    def _get_os_version(self) -> OsVersion:
```

Comment:
It looks most subclasses have its implementation of `_get_os_version`. It doesn't need to implement it in the Posix class.

Reply:
used by SUSE and any other derived from Linux and won't have its own implementation. I see no harm having this in here.

Reply:
"no harm" is not a reason to keep the code. Leave it not implemented is more clean to a base class. If not, other people may be confused when the code is run, they will spend time to understand where it applies.

Reply:
You ignored the first line in my comment. It is being used by SUSE and ofcourse other OSes like NixOS